src/tracer.js

```diff
@@ -237,8 +240,7 @@ export default class Tracer {
             ctx.baggage = parent.baggage;
 
             parent.finalizeSampling();
-            this.processDeferredSampling(ctx, operationName, internalTags);
-            ctx.finalizeSampling();
+            this.handleUpsampling(parent, ctx, operationName, internalTags);
```

Comment:
I'm not convinced if simply removing finalizedSampling has the same effect. If the span is started with opName="GET" and upsampling decides to sample, this is going to be final regardless of changes to the operationName later. Previously `finalized` logic was only making it just before starting a new span (or serializing)